common/djangoapps/course_modes/models.py

```diff
@@ -202,8 +202,18 @@ def clean(self):
             raise ValidationError(
                 _(u"Professional education modes are not allowed to have expiration_datetime set.")
             )
-        if self.is_verified_slug(self.mode_slug) and self.min_price <= 0:
-            raise ValidationError(_(u"Verified modes cannot be free."))
+
+        mode_config = settings.COURSE_ENROLLMENT_MODES.get(self.mode_slug, {})
+        min_price_for_mode = mode_config.get('min_price', 0)
+        if self.min_price < min_price_for_mode:
```

Comment:
Note that this will enforce constraints on course modes that did exist previously (e.g. all course mode prices must be non-negative now). I am assuming we do not want to support negative course modes anyway, but I'm leaving this here in case I'm mistaken.

Reply:
Yes, I don't think we'll be *paying* students to take a course...

Reply:
There goes my get-rich-quick scheme.